core/src/css/link.ionic.scss

```diff
@@ -0,0 +1,120 @@
+@import "../foundations/ionic.vars";
+
+// Link: Shared Styles (Standalone & Underline)
+// -------------------------------------------------------------------------------
+@mixin link-shared {
+  display: inline-flex;
+
+  align-items: center;
+
+  gap: 4px;
+
+  transition: color 0.2s ease-in-out;
+
+  font-weight: 400;
+
+  text-decoration-color: inherit;
+
+  text-underline-offset: 2px;
+
+  cursor: pointer;
+
+  // Link: Visited
+  // -------------------------------------------------------------------------------
+
+  &:visited {
+    color: $ionic-color-info-500;
+  }
+}
+
+// Link: Standalone
+// -------------------------------------------------------------------------------
+
+@mixin ion-link {
+  @include link-shared;
+
+  // Link: Standalone - Hover
+  // -------------------------------------------------------------------------------
+
+  @media (any-hover: hover) {
+    &:hover {
+      text-decoration: underline;
+    }
+  }
+
+  color: $ionic-color-info-400;
+
+  text-decoration: none;
+
+  // Link: Standalone - Focus
+  // -------------------------------------------------------------------------------
+
+  &:focus,
+  &.ion-focused {
+    outline: 2px solid $ionic-color-primary-100;
+    outline-offset: 2px;
+
+    text-decoration: underline;
+  }
+
+  // Link: Standalone - Active
+  // -------------------------------------------------------------------------------
+
+  &:active,
+  &.ion-activated {
+    color: $ionic-color-info-500;
+
+    text-decoration: underline;
+  }
+}
+
+a.ion-link,
+:not(a).ion-link > a {
```

Comment:
The design document states:

> Devs should be able to apply the classes to any element. (In other words, the styles should not be scoped to specific elements, like <a>.)

Should `a` be removed from the first selector? Or do we want to specify that the classes will only work on `element > a` or `a` elements?

Reply:
The link global classes should only be used to style a semantic anchor element. The design document was trying to communicate that the styles would be global CSS classes instead of only element-based styles on an anchor element, but worded it in a way that implied additional requirements.

All design requirements/visual styles are specific to an anchor element (visited styling, etc.) that are not applicable to attaching link styling to a non-anchor element. 

Refactoring the styles to only a CSS class would prevent usage at a parent element level, as the link styling would apply to all text content of the parent element and children. It would also require style resets on nested anchor tags to prevent the browser default styles. The current approach requires none of that and only applies to the anchor.

Reply:
Makes sense!